delta/delta_1.5.6_kek_model_update_2008_2019.sql

```diff
@@ -0,0 +1,424 @@
+
+ ALTER TABLE qgep_od.damage_manhole ADD COLUMN distance  decimal(7,2) ;
+COMMENT ON COLUMN qgep_od.damage_manhole.distance IS 'yyy_Länge von Rohranfang bis zur Feststellung bzw. Länge ab Oberkante Deckel (siehe Richtlinie Absatz 3.1.1.) in m mit zwei Nachkommastellen / Länge von Rohranfang bis zur Feststellung bzw. Länge ab Oberkante Deckel (siehe Richtlinie Absatz 3.1.1.) in m mit zwei Nachkommastellen / Longueur entre le début de la canalisation resp. longueur entre le bord supérieur du couvercle et l’observation (cf. paragraphe 3.1.1.) en m avec deux chiffres après la virgule.';
+
+-- Copy data from Superclass to Subclass
+
+UPDATE qgep_od.damage_manhole dm
+SET 
+	quantification1 = dg.quantification1,
+	quantification2 = dg.quantification2,
+	damage_begin = dg.damage_begin,
+	damage_end = dg.damage_end,
+	distance = dg.distance
+	
+FROM qgep_od.damage dg
+WHERE  dm.obj_id = dg.obj_id; 
+
+
+-- delete columns in qgep_od.damage
+
+ALTER TABLE qgep_od.damage
+DROP COLUMN IF EXISTS quantification1,
+DROP COLUMN IF EXISTS quantification2,
+DROP COLUMN IF EXISTS damage_begin,
+DROP COLUMN IF EXISTS damage_end,
+DROP COLUMN IF EXISTS distance;
+
+
+-- Adding further changes to attributes and values
+-- 1.5.6.2_kek_update_attribute_maintenance_event_identifier
+
+ALTER TABLE qgep_od.maintenance_event ALTER COLUMN  identifier TYPE varchar(41) ;
+COMMENT ON COLUMN qgep_od.maintenance_event.identifier IS '';
+
+-- 1.5.6.3_kek_update_attribute_maintenance_event_result
+
+ALTER TABLE qgep_od.maintenance_event ALTER COLUMN  result TYPE varchar(255) ;
+COMMENT ON COLUMN qgep_od.maintenance_event.result IS 'Result or important comments for this event / Resultat oder wichtige Bemerkungen aus Sicht des Bearbeiters / Résultat ou commentaire importante de l''événement';
+
+-- 1.5.6.4_kek_add_vl_file_upddate_attribute_object
+
+ALTER TABLE qgep_od.file ALTER COLUMN  object TYPE varchar(16) ;
+COMMENT ON COLUMN qgep_od.file.object IS 'yyy_Objekt-ID (OID) des Datensatzes zu dem die Datei gehört';
+
+-- 1.5.6.5_kek_add_vl_file_kind_add_new_values
+
+ INSERT INTO qgep_vl.file_kind (code, vsacode, value_en, value_de, value_fr, value_it, value_ro, abbr_en, abbr_de, abbr_fr, abbr_it, abbr_ro, active) VALUES (9146,9146,'digital_video','digitales_Video','video_numerique', 'video_digitale', 'rrr_digitales_Video', '', '', '', '', '', 'true');
+ INSERT INTO dictionary_od_values (code, vsacode, value_en, value_de, value_fr, value_it, value_ro, abbr_en, abbr_de, abbr_fr, abbr_it, abbr_ro, active) VALUES (9146,9146,'digital_video','digitales_Video','video_numerique', 'video_digitale', 'rrr_digitales_Video', '', '', '', '', '', 'true');
```

Comment:
As far as I remember, there's already an entry for digital_video, but spelled differently from the norm. If so, we should update instead of inserting

Reply:
To maintain the model so that you can see the changes also later, if a value changes during a model upgrade (patch or new release) it gets a new value ID. So I suggest to follow this principle als with the vl_* tables. You therefore have to add the new value ID, then replace old values with new values and then set the old value ID to active=false - see also below.